test/remote/gateways/remote_redsys_test.rb

```diff
@@ -18,6 +18,17 @@ def test_successful_purchase
     assert_equal 'Transaction Approved', response.message
   end
 
+  def test_successful_purchase_threeds2
+    response = @gateway.purchase(100, @credit_card, @options.merge(
+      three_d_secure: {
+        xid: '33ds22'
```

Comment:
Could you try exercising more 3DS fields here, [similar to the Adyen remote tests](https://github.com/activemerchant/active_merchant/blob/master/test/remote/gateways/remote_adyen_test.rb#L336-L343)?

We may also want to consider breaking this out into two separate remote tests:

- 3DS1
  - xid
  - cavv
  - eci
- 3DS2
  - threeDSServerTransID
  - dsTransID
  - authenticacionValue
  - protocolVersion
  - authenticacionMethod
  - authenticacionType
  - authenticacionFlow